course/include/isometry/isometry.hpp

```diff
@@ -10,15 +10,64 @@
 #include <initializer_list>
 #include <iostream>
 #include <sstream>
+#include <stdexcept>
 #include <string>
 
 namespace ekumen {
 
 namespace math {
 
-/*
- * your code here
- */
+class Vector3 {
+ public:
+  Vector3(const double x, const double y, const double z);
+  Vector3(std::initializer_list<double> list);
+  Vector3();
+  ~Vector3() { };
+
+  double operator[](const int i) const;
+  double& operator[](const int i);
+
+  double x() const { return this->x_; }
```

Comment:
It's not necessary to write `this->` before member variables. The compiler will implicitly add it on its own. Please remove all uses of `this`.